proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/logging/ProxyNodeLoggingFilter.java

```diff
@@ -29,25 +30,24 @@
 
     @Override
     public void filter(ContainerRequestContext requestContext) {
-        String istioTraceId = requestContext.getHeaderString(X_B3_TRACEID);
-        if (istioTraceId != null && !istioTraceId.isBlank()) {
-            MDC.put(X_B3_TRACEID, istioTraceId);
-        }
+        ofNullable(requestContext.getHeaderString(X_B3_TRACEID))
+            .filter(Predicate.not(String::isBlank)).ifPresent(traceId -> MDC.put(X_B3_TRACEID, traceId));
 
         ProxyNodeLogger.addContext(PROXY_NODE_JOURNEY_ID, getJourneyId(requestContext));
-        ProxyNodeLogger.addContext(REFERER, Optional.ofNullable(requestContext.getHeaderString(HttpHeaders.REFERER)).orElse(""));
-        Optional.ofNullable(requestContext.getUriInfo()).ifPresent(u -> ProxyNodeLogger.addContext(RESOURCE_PATH, u.getAbsolutePath().toString()));
-        Optional.ofNullable(requestContext.getMediaType()).ifPresent(m -> ProxyNodeLogger.addContext(INGRESS_MEDIA_TYPE, m.toString()));
-        ProxyNodeLogger.info(MESSAGE_INGRESS);
+        ProxyNodeLogger.addContext(REFERER, ofNullable(requestContext.getHeaderString(HttpHeaders.REFERER)).orElse(""));
 
+        ofNullable(requestContext.getUriInfo()).ifPresent(u -> ProxyNodeLogger.addContext(RESOURCE_PATH, u.getAbsolutePath().toString()));
+        ofNullable(requestContext.getMediaType()).ifPresent(m -> ProxyNodeLogger.addContext(INGRESS_MEDIA_TYPE, m.toString()));
+
+        ProxyNodeLogger.info(MESSAGE_INGRESS);
         MDC.remove(INGRESS_MEDIA_TYPE.name());
     }
 
     @Override
     public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
-        Optional.ofNullable(responseContext.getLocation()).ifPresent(uri -> ProxyNodeLogger.addContext(EGRESS_LOCATION, uri.toString()));
```

Comment:
I remember being told that we favoured the fully-qualified imports i.e. `Optional.ofNullable`. We should discuss this  at dev huddle.

Reply:
Yes we do most of the time